Sources/JWT/Signers/RSASigner.swift

```diff
@@ -26,67 +26,117 @@ extension HashMethod {
     }
 }
 
-public struct RS256: RSASigner {
-    public let key: Bytes
+public typealias CRSAKey = UnsafeMutablePointer<RSA>
+
+public enum RSAKey {
+    case `public`(CRSAKey)
+    case `private`(CRSAKey)
+
+    public init(_ rawKey: Bytes) throws {
```

Comment:
I think it would also be safe to run both conversions in a single withUnsafeBufferPointer closure. That would also reduce duplication.